packages/babel-parser/src/parser/node.js

```diff
@@ -47,6 +48,46 @@ class Node implements NodeBase {
     return newNode;
   }
 }
+const NodePrototype = Node.prototype;
+
+function clonePlaceholder(node: any): any {
+  return cloneIdentifier(node);
+}
+
+export function cloneIdentifier(node: any): any {
+  // We don't need to clone `typeAnnotations` and `optional`: because
+  // cloneIdentifier is only used in object shorthand and named import/export.
+  // Neither of them allow type annotations after the identifier or optional identifier
+  const { type, start, end, loc, range, extra, name } = node;
+  const cloned = Object.create(NodePrototype);
```

Comment:
We can use plain object after we remove `__clone` method from the AST node.

Reply:
why do you need to close? You will experience more than 80% perf boost if you avoid clone and instead rewrite lexer and parser. And also all the property access you are using kills perf.